src/java.base/share/classes/java/net/URLEncoder.java

```diff
@@ -161,7 +163,7 @@ public static String encode(String s) {
         try {
             str = encode(s, dfltEncName);
         } catch (UnsupportedEncodingException e) {
```

Comment:
Perhaps a separate cleanup, the Charset should be cached, not just the name and use the `encode(s, charset)` method.